src/arviz_plots/plots/evolutionplot.py

```diff
@@ -0,0 +1,449 @@
+    aes_map.setdefault("mean", plot_collection.aes_set)
+    aes_map.setdefault("mean", ["linestyle", "-"])
+    aes_map.setdefault("sd", plot_collection.aes_set)
+    aes_map.setdefault("sd", ["linestyle", "-"])
+    aes_map.setdefault("divergence", {"overlay"})
+    if labeller is None:
+        labeller = BaseLabeller()
+
+    # compute and add ess evolution subplots for 'bulk' and 'tail'
+
+    n_draws = distribution.sizes["draw"]
+    n_samples = n_draws * distribution.sizes["chain"]
+    plot_bknd = import_module(f".backend.{backend}", package="arviz_plots")
+    default_bulk_color = plot_bknd.get_default_aes("color", 2, {})[0]
+    default_tail_color = plot_bknd.get_default_aes("color", 2, {})[1]
+
+    bulk_kwargs = copy(plot_kwargs.get("bulk", {}))
+
+    if bulk_kwargs is not False:
+        # step 2
+        bulk_dims, bulk_aes, bulk_ignore = filter_aes(plot_collection, aes_map, "bulk", sample_dims)
+        first_draw = distribution.draw.values[0]
+        ylabel = "{}"
+        xdata = np.linspace(n_samples / n_points, n_samples, n_points)
+        draw_divisions = np.linspace(n_draws // n_points, n_draws, n_points, dtype=int)
+        ess_y_dataset = xr.concat(
+            [
+                distribution.sel(draw=slice(first_draw + draw_div)).azstats.ess(
+                    dims=bulk_dims,
+                    method="bulk",
+                    relative=relative,
+                    **stats_kwargs.get("ess_bulk", {}),
+                )
+                for draw_div in draw_divisions
+            ],
+            dim="ess_dim",
+        )
+
+        # ----------------------------
+
+        xdata = np.linspace(0, n_samples, n_points)
+
+        print(f"\n ess_y_dataset = {ess_y_dataset}")
+
+        # converting xdata into a xr dataarray
+        xdata_da = xr.DataArray(xdata, dims="ess_dim")
+        # print(f"\n xdata_da ={xdata_da}")
+
+        # broadcasting xdata_da to match shape of each variable in ess_y_dataset and
+        # creating a new dataset from dict of broadcasted xdata
+        xdata_dataset = xr.Dataset(
+            {var_name: xdata_da.broadcast_like(da) for var_name, da in ess_y_dataset.items()}
+        )
+        # print(f"\n xdata_dataset = {xdata_dataset}")
+
+        # concatenating xdata_dataset and ess_y_dataset along plot_axis
+        ess_dataset = xr.concat([xdata_dataset, ess_y_dataset], dim="plot_axis").assign_coords(
+            plot_axis=["x", "y"]
+        )
+        print(f"\n distribution = {distribution!r}")
+        print(f"\n ess_dataset = {ess_dataset!r}")
+
+        # step 4
+        if "color" not in bulk_aes:
+            bulk_kwargs.setdefault("color", default_bulk_color)
+
+        # step 5
+        plot_collection.map(
+            scatter_xy, "bulk", data=ess_dataset, ignore_aes=bulk_ignore, **bulk_kwargs
+        )
+        plot_collection.map(
+            line_xy, "bulk", data=ess_dataset, ignore_aes=bulk_ignore, **bulk_kwargs
+        )
```

Comment:
These are two different visual elements which should have different names and consequently different kwargs too. It should be possible to remove one but not the other, and it should be possible to change the marker of the scatter for example, but if we use the same kwargs, these are then passed to the line plotting function which doesn't accept the marker kwarg.